### PR TITLE
[eBPF] Memory optimization for the profiler

### PR DESCRIPTION
**Pull request by yinjiping**
Using regular expressions, we match process names to establish symbol table caching for specific processes, rather than enabling symbol caching for all processes. This approach aims to reduce memory usage.

For example:

`^(java|nginx|profiler|telegraf|mysqld|socket_tracer|.*deepflow.*)$`

This regex pattern matches the process names: "java", "nginx", "profiler", "telegraf", "mysqld", "socket_tracer", and any process containing the substring "deepflow".

You can modify the regular expression according to your needs using the `set_profiler_regex()` interface.
By using this interface, you can customize the regular expression to match specific process names that you want to establish symbol table caching for, providing flexibility and control over which processes will have symbol caching enabled. This allows you to finetune the memory usage and profiling behavior of the profiler according to your requirements.

The default expression is empty (''), indicating that no profile data will be generated.

### This PR is for:

- Agent

#### Affected branches
- main

### CODE REVIEW

File: agent/src/ebpf/samples/rust/profiler/src/main.rs [hunk 1]
```diff
@@ -112,6 +115,20 @@ fn main() {
...
         }
 
+       if running_socket_tracer(
+            socket_trace_callback, /* 回调接口 rust -> C */
+            1,                     /* 工作线程数，是指用户态有多少线程参与数据处理 */
+            64,                   /* 内核共享内存占用的页框数量, 值为2的次幂。用于perf数据传递 */
+            65536,                 /* 环形缓存队列大小，值为2的次幂。e.g: 2,4,8,16,32,64,128 */
+            524288, /* 设置用于socket追踪的hash表项最大值，取决于实际场景中并发请求数量 */
+            524288, /* 设置用于线程/协程追踪会话的hash表项最大值。*/
+            520000, /* socket map表项进行清理的最大阈值，当前map的表项数量超过这个值进行map清理操作 */
```
Comment (sharang) [inline agent/src/ebpf/samples/rust/profiler/src/main.rs:125]: 注释用英文吧

File: agent/src/ebpf/samples/rust/socket-tracer/src/main.rs [hunk 2]
```diff
@@ -409,7 +409,7 @@ fn main() {
         //}
 
         //set_profiler_regex(
-        //    CString::new("^(java|nginx|profiler|telegraf|mysqld|.*deepflow.*)$".as_bytes())
+        //    CString::new("^(java|nginx|profiler|telegraf|mysqld|.*deepflow.*|socket_tracer)$".as_bytes())
```
Comment (sharang) [inline agent/src/ebpf/samples/rust/socket-tracer/src/main.rs:412]: 默认值为空吧，表示不做任何 continuous profile 。

后面可以在 agent 配置中提醒配置这些来采集常见的进程，但当前阶段默认还是先关闭。

File: agent/src/ebpf/user/perf_profiler.c [hunk 3]
```diff
@@ -752,7 +739,7 @@ static int create_profiler(struct bpf_tracer *tracer)
 	 * Use of void* is inherited from the BCC library. */
 	create_and_init_symbolizer_caches();
 
-	set_profiler_regex(".*");
+	set_profiler_regex("");
```
Comment (nrjatyunshan) [inline agent/src/ebpf/user/perf_profiler.c:751]: "" 会匹配所有字符串. 跟 ".*" 效果是一样的

Comment (yinjiping) [reply agent/src/ebpf/user/perf_profiler.c:751]: 是的，代码中重新约定了：
```
int set_profiler_regex(const char *pattern)
{
        if (profiler_tracer == NULL) {
                ebpf_warning(LOG_CP_TAG "The 'profiler_tracer' has not been created yet."
                             " Please use start_continuous_profiler() to create it first.\n");
                return (-1);
        }

        /*
         * During the data processing, the thread responsible for matching reads the
         * regular expression, while the thread handling the regular expression upd-
         * ates is different. Synchronization is implemented to ensure protection and
         * coordination between these two threads.
         */
        tracer_reader_lock(profiler_tracer);
        if (*pattern == '\0') {
                regex_existed = false;
                ebpf_warning(LOG_CP_TAG "Set 'profiler_regex' pattern : '', an empty"
                             " regular expression will not generate any stack data."
                             "Please configure the regular expression for profiler.\n");
                tracer_reader_unlock(profiler_tracer);
                return (0);
        }
```

**Review by sharang — APPROVED** [review]

**Pull request merged**